deebotozmo/cli.py

```diff
@@ -0,0 +1,439 @@
+    await lock.acquire()
+    vacbot.bot.events.lifespan.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+@cli.command(name="getrooms", help="Get saved rooms")
+@coro
+async def get_rooms() -> None:
+    """Click subcommand that returns saved room."""
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+
+    async def on_rooms(event: RoomsEvent) -> None:
+        for room in event.rooms:
+            print(str(room.id) + " " + str(room.subtype))
+        lock.release()
+
+    listener = vacbot.bot.map.events.rooms.subscribe(on_rooms)
+    await lock.acquire()
+    vacbot.bot.map.events.rooms.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+@cli.command(
+    name="exportlivemap",
+    help='Get robot map and save it [filepath ex: "/folder/livemap.png"',
+)
+@click.argument("filepath", type=click.STRING, required=True)
+@coro
+async def export_live_map(filepath: str) -> None:
+    """Click subcommand that returns the live map."""
+    vacbot = DoLogin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+
+    async def on_map(_: MapEvent) -> None:
+        with open(filepath, "wb") as file:
+            file.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+        lock.release()
+
+    listener = vacbot.bot.events.map.subscribe(on_map)
+    await lock.acquire()
+    vacbot.bot.events.map.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+
+class DoLogin:
+    """DoLogin communicates with the bot to provide self.bot."""
+
+    def __init__(self) -> None:
+        """Init DoLogin."""
+        # self.session = None
+        # self.auth = None
+        # self.devices_ = None
+        # self.bot = None
+        # self.mqtt = None
+        return
+
+    async def run(self) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        """Communicate with Deebot."""
+        config = read_config()
+
+        self.session = aiohttp.ClientSession()
+
+        self.api = EcovacsAPI(
+            self.session,
+            config["device_id"],
+            config["email"],
+            config["password_hash"],
+            continent=config["continent"],
+            country=config["country"],
+            verify_ssl=bool(config["verify_ssl"]),
+        )
+
+        if not config_file_exists():
+            click.echo(
+                "Not logged in. "
+                + f"Do '{os.path.basename(sys.argv[0])} createconfig' first."
+            )
+            sys.exit(1)
+
+        await self.api.login()
+
+        self.devices_ = await self.api.get_devices()
+
+        self.auth = await self.api.get_request_auth()
+        self.bot = VacuumBot(
+            self.session,
+            self.auth,
+            self.devices_[0],
+            continent=config["continent"],
+            country=config["country"],
+            verify_ssl=bool(config["verify_ssl"]),
+        )
+
+        self.mqtt = EcovacsMqtt(
```

Comment:
mqtt can be removed as it is used for events send out by the robot asynchronous. As you unsubscribe straight away this is not required or did you get some events only over mqtt?

Reply:
Yes if I removed mqtt I no longer get statuses

Reply:
Which status?

Reply:
the vacuum state (whether its docking, etc)

Reply:
With my last suggestion it should be possible to remove mqtt.